File: jobbazar_mobile/lib/provider/models/job.dart
class Job {
  final int id;
  final String title;
  final String description;
  final String location;
  final String type;
  final int employerId;
  final double salary;
  final String company;

  Job({required this.id, required this.title, required this.description, required this.location, required this.type, required this.employerId, required this.salary, required this.company});

factory Job.fromJson(Map<String, dynamic> json) {
    return Job(
      id: json['id'] ?? -1,
      title: json['title'] ?? 'Unknown Title',
      description: json['description'] ?? 'No Description',
      location: json['location'] ?? 'Unknown Location',
      type: json['type'] ?? 'Unknown Type',
      employerId: json['employerId'] ?? -1,
      salary: (json['salary'] as num?)?.toDouble() ?? 0.0,
      company: json['company'] ?? 'Unknown Company',
    );
  } 
}

File: backend/src/main/java/com/siyam/jobListing/Jobs/Model/Job.java
package com.siyam.jobListing.Jobs.Model;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(name = "jobs")
public class Job {
    public enum JobType {
        FULL_TIME,
        PART_TIME,
        CONTRACT,
        INTERNSHIP
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "company")
    private String company;

    @Column(name = "location")
    private String location;

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    private JobType type;

    @Column(name = "salary")
    private Double salary;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonProperty("employer")
    private User employer;

    @OneToMany(mappedBy = "job", cascade = CascadeType.ALL, orphanRemoval = true)
//    @JsonBackReference
    private List<Application> applications;

}
